samza-api/src/main/java/org/apache/samza/operators/MessageStream.java

```diff
@@ -113,12 +113,13 @@
    *
    * @param otherStream the other {@link MessageStream} to be joined with
    * @param joinFn the function to join messages from this and the other {@link MessageStream}
+   * @param ttlMs the ttl in ms for retaining messages in each stream
    * @param <K> the type of join key
    * @param <OM> the type of messages in the other stream
    * @param <RM> the type of messages resulting from the {@code joinFn}
    * @return the joined {@link MessageStream}
    */
-  <K, OM, RM> MessageStream<RM> join(MessageStream<OM> otherStream, JoinFunction<K, M, OM, RM> joinFn);
+  <K, OM, RM> MessageStream<RM> join(MessageStream<OM> otherStream, JoinFunction<K, M, OM, RM> joinFn, long ttlMs);
```

Comment:
I wonder if this API can take a `Duration` instead. 
* It's slightly consistent with the rest of our APIs for windows.
* `Duration.ofHours(2)` is arguably more readable(?) than 2 * 60 * 60 * 1000 milliseconds